unimus-backup-exporter.sh

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env bash
+
+#This is a Unimus to Git API to export your backups to your Git Repo
+
+
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+backup_dir=$script_dir/backups
+
+
+
+#Imports from env
+set -a # automatically export all variables
+source unimus-backup-exporter.env
+set +a
+
+
+#HashTable for all devices
+declare -A devices
+log=$script_dir/unimus-backup-exporter.log
+
+
+function echoGreen() { echo -e "${green}$1${reset}"; }
+function echoYellow() { echo -e "${yellow}$1${reset}"; }
+function echoRed() { echo -e "${red}$1${reset}"; }
+
+
+#This function will do a get request 
+function unimus_get () {
+	local get_request=$(curl -s -H "Accept: application/json" -H "Authorization: Bearer $unimus_api_key" "$unimus_server/api/v2/$1")
+	echo "$get_request"
+}
+
+#Verify's Server is online
+function unimus_status_check () {
+	local get_status=$(unimus_get "health")
+	local status=$(jq -r '.data.status' <<< $get_status)
+	echo "$status"
+}
+
+#Decodes and Saves File
+function save_file() {
+	local address=${devices[$1]}
+	local date="$2"
+	if [[ $4 == "TEXT" ]]; then
+		local type="txt"
+	elif [[ $4 == "BINARY" ]]; then
+		local type ="bin"
+	fi
+
+	if ! [ -d "backups" ]; then
+		mkdir backups
+	fi
+
+	if ! [ -d "$backup_dir/$address - $1" ]; then
+		mkdir $backup_dir/$address\ \-\ $1
+	fi
+	if ! [ -e $backup_dir/$address\ -\ $1/Backup\ $address\ $date\ $1.$type ]; then
+		base64 -d <<< $3 > $backup_dir/$address\ -\ $1/Backup\ $address\ $date\ $1.$type
+	fi
+}
+
+function get_allDevices (){
+	echoGreen "Getting Device Information" >> $log
+	for ((i=0; ; i+=1)); do
```

Comment:
`while true` would be better

Reply:
I was incorrect here. We use the for loop to traverse through pages and each json entry.